urlResolvers/github.js

```diff
@@ -0,0 +1,42 @@
+const URLResolver = require('./urlResolver');
+const GitHubAPIHandler = require('../sourceHandlers/github');
+
+const stripHttp = /^(http(s)?:\/\/)?(www.)?/;
+const urlRegex = /^github\.com\/([\w,-]+)\/([\w,-]+)/;
+
+const apiBaseUrl = 'https://api.github.com/';
+
+/**
+ * Creates the supported source handlers for a given URL.
+ * @class
+ * @memberof urlResolvers
+ * @augments urlResolvers.URLResolver
+ */
+class GitHubResolver extends URLResolver {
+  static getSourceHandlers(url) {
+    const sourceHandlers = [];
+    const repoInfo = this._parseURL(url);
+    if (repoInfo != null) {
+      const repoPath = `${repoInfo.owner}/${repoInfo.repoName}`;
+
+      sourceHandlers.push(new GitHubAPIHandler(apiBaseUrl, repoPath));
+    }
+
+    return sourceHandlers;
+  }
+
+  static _parseURL(url) {
```

Comment:
I personally prefer to implement before execution, when possible. I think it makes it a bit easier to understand when reading through new code, since you've already read how a function works when you get to its use.